Linux Athena FRCDriverStation.cpp build error on GCC 11.2

**Describe the bug**
Build error in `hal/src/main/native/athena/FRCDriverStation.cpp` when using a custom roboRIO toolchain with GCC 11.2

**Expected behavior**
Successful build as seen with GCC 7.3

**Screenshots**
```
> Task :hal:compileHalBaseLinuxathenaReleaseStaticLibraryHalBaseLinuxathenaReleaseStaticLibraryAthenaCpp
/home/ryan/build/allwpilib/hal/src/main/native/athena/FRCDriverStation.cpp: In function 'char* HAL_GetJoystickName(int32_t)':
/home/ryan/build/allwpilib/hal/src/main/native/athena/FRCDriverStation.cpp:305:17: error: 'char* strncpy(char*, const char*, size_t)' output truncated before terminating nul copying as many bytes from a string as its length [-Werror=stringop-truncation]
  305 |     std::strncpy(name, joystickDesc.name, len);
      |     ~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/ryan/build/allwpilib/hal/src/main/native/athena/FRCDriverStation.cpp:303:29: note: length computed here
  303 |     size_t len = std::strlen(joystickDesc.name);
      |                  ~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~
cc1plus: all warnings being treated as errors
```

**Desktop (please complete the following information):**
 - OS: WSL Ubuntu 18.04
 - Java version: Distro provided OpenJDK 11
 - C++ version: 17

**Additional context**
Toolchain was built by me and a custom version of native-utils was used to plug in my toolchain.

--- a/hal/src/main/native/athena/FRCDriverStation.cpp
+++ b/hal/src/main/native/athena/FRCDriverStation.cpp
@@ -302,7 +302,7 @@ char* HAL_GetJoystickName(int32_t joystickNum) {
   } else {
     size_t len = std::strlen(joystickDesc.name);
     char* name = static_cast<char*>(std::malloc(len + 1));
-    std::strncpy(name, joystickDesc.name, len);
+    std::memcpy(name, joystickDesc.name, len);
     name[len] = '\0';
     return name;
   }

